### DISCUSSION TITLE
Clear transmitter queue is broken

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2088] [General]
I am running: 
2022.04.22
dae25a9 

I wanted to test the stop sensor command with the latest release.
After stopping a G6 (Firefly) sensor, I saw the message "Now start your sensor or check Settings".
So, I wanted to undo the stop command and cleared the queue as I have done many times before.

The queue was cleared and the transmitter never received the stop command as BlueJay continues to work fine.
But, xDrip refuses to show readings.
I shut down the phone and established connectivity using an older phone and got proper operation and readings on the old phone.

Going back to the original phone, I still get no readings.  The system status page shows OK 9.8 days.  So, the connectivity is OK.  
I think something is wrong with how xDrip reacts to the stop commend in the latest Nightly release.

**Comment by Navid200** [comment]
I went back to the April 20, 2022 release.
But, it still refuses to show readings.  And, the start sensor option remains available in the menu.
I have never seen this before.

I will be starting a sensor today and I was hoping to test the latest release.  But, I'm not sure what to do now because I am not happy with this behavior at all.

The clear queue command always cancelled everything.  And establishing connectivity to a working session in progress always restored 
operation with no need to start a sensor.

**Comment by Navid200** [comment]
I went back to the latest release (April 22, 2022) and tapped on start.  The readings are back.
I guess I can live with this and I can update my instructions.

But, clearing the queue before the stop command is sent to the transmitter should not put xDrip in a non-functioning mode.

**Reply by jamorham** [reply]
Erasing the queue command is going to put it in an inconsistent state because it expects that queue command to stop the sensor session bringing it in line with xDrips already stopped sensor session. I don't think it would be non-functioning, you just have to select Start Sensor should restore display of data.

**Reply by Navid200** [reply]
But, this is new.

The consequence is that no matter what the cause of the problem is, the user will have to tap on start in order to confirm if there is a session in progress or not.

I am curious why this change, the requirements to tap on start, is necessary to receive readings from the session in progress.
What was wrong with how it behaved before?

**Reply by Navid200** [reply]
Let me ask this question differently.

If we allow a user to get readings from a session in progress without having to tap on start, what could go wrong?
